### PR TITLE
LF-2776: New completion date component for task completion flow resolve conflict

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

This PR is to complete @mickenorlen's PR: https://github.com/LiteFarmOrg/LiteFarm/pull/2536
I created the branch from his branch, merged integration branch and resolved conflicts.

Jira link: https://lite-farm.atlassian.net/browse/LF-2776

### CODE REVIEW

**Review by Pazzolo — DISMISSED** [review]

File: packages/webapp/src/components/Task/TaskComplete/index.jsx [hunk 1]
```diff
@@ -26,23 +27,38 @@ export default function PureTaskComplete({
...
-  const COMPLETE_DATE = 'complete_date';
+  const DATE_CHOICE = 'date_choice';
+  const ANOTHER_DATE = 'date_another';
 
   const { t } = useTranslation();
 
+  // Prepare dates
+  const date_due = getDateInputFormat(persistedFormData.due_date);
+  const date_today = getDateInputFormat();
+  const dueDateDisabled = date_due >= date_today;
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Task/TaskComplete/index.jsx:38]: @SayakaOno This is disabling the ability to select the first option which I don't think is desired.

File: packages/webapp/public/locales/en/translation.json [hunk 2]
```diff
@@ -1670,7 +1670,7 @@
       "WHICH_DATE": "Which date?",
       "DATE_ORIGINAL": "The original due date",
       "DATE_TODAY": "Today",
-      "DATE_ANOTHER": "Another",
+      "DATE_ANOTHER": "Another date",
```
Comment (Duncan-Brain) [inline packages/webapp/public/locales/en/translation.json:1673]: The other translations should be reset to MISSING

File: packages/webapp/src/components/Form/Radio/index.jsx [hunk 3]
```diff
@@ -18,6 +19,9 @@ const Radio = ({
   ...props
 }) => {
   const name = hookFormRegister?.name ?? props?.name;
+  const pill = props?.pill;
+  const checked = props?.checked;
```
Comment (Duncan-Brain) [inline packages/webapp/src/components/Form/Radio/index.jsx:23]: same I think we could add this to explicit props?

**Comment by Duncan-Brain** [comment]
@SayakaOno and @mickenorlen 

Looks good to me, only dealbreaker is the original due date logic -- I don't think we want any:
Screenshot 2023-04-12 at 7 04 39 PM

Also the translations: any updates to english means we need to invalidate the others by replacing with MISSING.

**Comment by SayakaOno** [comment]
@Duncan-Brain Thank you for the comments!
1. another date default value - removed
2. translations - fixed

- Radio component - please check if I understand what you meant correctly
- Several changes were made (including making the radio options' "value" static) to enable all options
- The request for this task was to make the behaviour consistent with `AbandonTask`, so I removed validations although they were good.

File: packages/webapp/src/components/Task/TaskComplete/index.jsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Task/TaskComplete/index.jsx]: Removed!

File: packages/webapp/src/components/Form/Radio/index.jsx [hunk 4]
```diff
@@ -18,6 +19,9 @@ const Radio = ({
   ...props
 }) => {
   const name = hookFormRegister?.name ?? props?.name;
+  const pill = props?.pill;
```
[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/components/Form/Radio/index.jsx]: Now it's destructured.

File: packages/webapp/src/components/Form/Radio/index.jsx (hunk 3 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Form/Radio/index.jsx]: Now it's destructured too!

File: packages/webapp/src/components/Task/TaskComplete/index.jsx (hunk 1 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/Task/TaskComplete/index.jsx]: @SayakaOno 

I think I was wrong to suggest removing this even though abandon task does it... If you think it is correct I would leave it in.

**Comment by Duncan-Brain** [comment]
@SayakaOno @mickenorlen 

I think I was wrong to suggest removing the "original due date" logic. Even though it is meant to match the abandon task I think the validation is good. I was thrown off by the unselectable radio button -- I think the best solution would be to not show the option at all.

File: packages/webapp/src/components/Task/TaskComplete/index.jsx (hunk 1 above)
Comment (SayakaOno) [reply packages/webapp/src/components/Task/TaskComplete/index.jsx]: I don't think the option is disabled in AbandonTask. Could you double check?
`/packages/webapp/src/components/Task/AbandonTask/index.jsx`

**Comment by SayakaOno** [comment]
I added validation to both `TaskComplete` and `AbandonTask`.

File: packages/webapp/src/components/Task/TaskComplete/index.jsx (hunk 1 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/Task/TaskComplete/index.jsx]: @SayakaOno -- no it doesn't disable it you are right. But it seems like they both _should_ prevent abandoning or completing on a future date. Even though it is not currently in place it seems like that is desired functionality to disable(or hide) the option if the original due date is at some point in the future.

**Comment by SayakaOno** [comment]
We decided to disable the original due date option when it is a future date.

**Review by Pazzolo — APPROVED** [review]

**Review by Duncan-Brain — APPROVED** [review]
Thanks for this and approved. 

Possible future question: If todays date = original due date maybe we should not disable 😂 ? Not necessary for this pr tho.

**Comment by SayakaOno** [comment]
@Duncan-Brain Thank you Duncan! That's what I was wondering! Users can use the second option for now...😂

**Pull request merged**